components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/IdentityRecoveryServiceComponent.java

```diff
@@ -517,4 +518,20 @@ protected void unsetApplicationManagementService(ApplicationManagementService ap
 
         IdentityRecoveryServiceDataHolder.getInstance().setApplicationManagementService(null);
     }
+
+    @Reference(
+            name = "workflowservice.default",
+            service = org.wso2.carbon.identity.workflow.mgt.WorkflowManagementService.class,
+            cardinality = ReferenceCardinality.MANDATORY,
```

Comment:
This change correctly sets the `ReferenceCardinality.MANDATORY`, which is good because it ensures the `workflowservice.default` OSGi service is always available and prevents runtime issues caused by a missing mandatory dependency.